prusti-specs/src/specifications/preparser.rs

```diff
@@ -285,7 +286,8 @@ pub struct Parser {
     /// will be expected.
     expected_operator: bool,
     /// A flag to denote that the next token must be an operator.
-    expected_only_operator: bool
+    expected_only_operator: bool,
+    parsing_pledge_with_lhs: bool
```

Comment:
```suggestion
    parsing_pledge_with_lhs: bool,
```

Could you also add a doc comment explaining this field.